build/test_stdio_snapshot.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Generic test script to verify workerd test output against stdio snapshot files
```

Comment:
It would have been helpful if the changes related to factoring out wd_stdio_snapshot_test were in a separate commit, it took me a while to understand what this file was doing here.

I'm still not quite sure why it's needed. Could you instead test the unload handler by having it set global variables, and then you send another request to the same isolate which verifies the values in those globals?

Testing by matching stdout feels like it may be brittle.